src/t_stream.c

```diff
@@ -91,6 +91,21 @@ void streamIncrID(streamID *id) {
     }
 }
 
+/* Set 'id' to be its predecessor streamID */
+void streamDecrID(streamID *id) {
+    if (id->seq == 0) {
+        if (id->ms == 0) {
+            /* Special case where 'id' is the first possible streamID... */
+            id->ms = id->seq = 0;
```

Comment:
don't you need to set both to UINT64_MAX?

Reply:
I'm not sure. I think there's a subtle but important difference between `0-0` and `UINT64_MAX-UINT64_MAX`: the first isn't a valid id, whereas the second is. So 0-0 is kind of a null iiuc - @guybe7?

Reply:
that depends how we wants to treat `(0` or `(0-0` as `end`... IMO we should indeed set the id to "null" so that the XRANGE command will return an empty array...

note that now we have some sort of weird behavior:
`XRANGE 18446744073709551615-18446744073709551615 +` will always return an empty array
`XRANGE (18446744073709551615-18446744073709551615 +` will return the entire stream

maybe we should just reply with error if user incrSteeamId or decrStreamId "wrap around"

Reply:
given the fact that whose are meaningless edge-cases i think an error is the right way to go

Reply:
i vote for error.
but with regards to the `streamDecrID` function being independent, i'd prefer it to do wrap around, like `streamIncrID` does.

Reply:
Agreed - resolved.